File: dotnet/src/adapter/driving/RequestAuthHelper.cs
using DddExample.Application;
using DddExample.Utility;

namespace DddExample.Adapter.Driving
{
    /// <summary>
    /// 请求 Auth Helper
    /// </summary>
    public class RequestAuthHelper
    {
        private readonly UserAuthService userAuthService;
        private readonly UserQueryHandler userQueryHandler;

        private static RequestAuthHelper? INSTANCE;

        public RequestAuthHelper(UserAuthService userAuthService, UserQueryHandler userQueryHandler)
        {
            this.userAuthService = userAuthService;
            this.userQueryHandler = userQueryHandler;
            INSTANCE = this;
        }

        /// <summary>
        /// 获取静态实例
        /// </summary>
       private static RequestAuthHelper GetInstance()
        {
            if (INSTANCE == null)
                throw new ControllerException("RequestAuthHelper 静态实例未初始化");
            return INSTANCE;
        }

        /// <summary>
        /// 获取登录用户Id，未登录返回空字符串
        /// </summary>
        public static string getLoginUserId(HttpRequest request)
        {
            string? accessToken = null;
            if (request.Headers.TryGetValue("Access-Token", out var values))
                accessToken = values.FirstOrDefault();

            if (accessToken == null)
                return string.Empty;

            return GetInstance().userAuthService.GetLoginUserId(accessToken);
        }

        /// <summary>
        /// 获取登录用户Id，未登录抛异常
        /// </summary>
        public static string RequireLoginUserId(HttpRequest request)
        {
            string userId = getLoginUserId(request);
            if (ValueUtility.IsBlank(userId))
                throw new NotLoginException();
            return userId;
        }

        /// <summary>
        /// 获取登录用户（管理员），失败抛异常
        /// </summary>
        public static UserDto RequireLoginUserAdmin(HttpRequest request)
        {
            string userId = RequireLoginUserId(request);
            UserDto? userDto = GetInstance().userQueryHandler.QueryById(userId);
            if (userDto == null || !userDto.IsAdmin)
                throw new NotLoginException();
            return userDto;
        }
    }
}

File: backend-example-python/src/adapter/driving/request_auth_helper.py
from starlette.requests import Request

from src.adapter.driving.not_login_exception import NotLoginException
from src.application.user_auth_service import UserAuthService
from src.application.user_dto import UserDto
from src.application.user_query_handler import UserQueryHandler
from src.ioc_container import ioc_container
from src.utility import value_utility

user_auth_service = ioc_container.resolve(UserAuthService)  # type: ignore
user_query_handler = ioc_container.resolve(UserQueryHandler)  # type: ignore


def get_login_user_id(request: Request) -> str:
    """获取登录用户Id，失败返回空字符串"""
    access_token = request.headers.get("Access-Token", "")
    return user_auth_service.get_login_user_id(access_token)


def require_login_user_id(request: Request) -> str:
    """获取登录用户Id，失败抛出异常"""
    user_id = get_login_user_id(request)
    if value_utility.is_blank(user_id):
        raise NotLoginException()
    return user_id


async def require_login_user_admin(request: Request) -> UserDto:
    """获取登录用户（管理员），失败抛出异常"""
    user_id = get_login_user_id(request)
    user_dto = await user_query_handler.query_by_id(user_id)
    if user_dto is None or not user_dto.is_admin:
        raise NotLoginException()
    return user_dto
